relay-server/src/actors/processor/span.rs

```diff
@@ -0,0 +1,418 @@
+//! Processor code related to standalone spans.
+#[cfg(feature = "processing")]
+use {std::error::Error, std::sync::Arc};
+
+use relay_dynamic_config::Feature;
+#[cfg(feature = "processing")]
+use {
+    chrono::{DateTime, Utc},
+    relay_base_schema::events::EventType,
+    relay_config::Config,
+    relay_dynamic_config::ErrorBoundary,
+    relay_dynamic_config::ProjectConfig,
+    relay_event_normalization::span::tag_extraction,
+    relay_event_schema::processor::{process_value, ProcessingState},
+    relay_event_schema::protocol::Span,
+    relay_metrics::{aggregator::AggregatorConfig, MetricNamespace, UnixTimestamp},
+    relay_pii::PiiProcessor,
+    relay_protocol::{Annotated, Empty},
+};
+
+use crate::{actors::processor::ProcessEnvelopeState, envelope::ItemType, utils::ItemAction};
+#[cfg(feature = "processing")]
+use {
+    crate::actors::processor::ProcessingError,
+    crate::envelope::{ContentType, Item},
+    crate::metrics_extraction::generic::extract_metrics,
+};
+
+pub fn filter(state: &mut ProcessEnvelopeState) {
+    let standalone_span_ingestion_enabled = state
+        .project_state
+        .has_feature(Feature::StandaloneSpanIngestion);
+    state.managed_envelope.retain_items(|item| match item.ty() {
+        ItemType::OtelSpan | ItemType::Span => {
+            if !standalone_span_ingestion_enabled {
+                relay_log::warn!("dropping span because feature is disabled");
+                ItemAction::DropSilently
+            } else {
+                ItemAction::Keep
+            }
+        }
+        _ => ItemAction::Keep,
+    });
+}
+
+/// Config needed to normalize a standalone span.
+#[cfg(feature = "processing")]
+#[derive(Clone, Debug)]
+pub struct NormalizeSpanConfig {
+    /// The time at which the event was received in this Relay.
+    pub received_at: DateTime<Utc>,
+    /// Allowed time range for transactions.
+    pub transaction_range: std::ops::Range<UnixTimestamp>,
+    /// The maximum allowed size of tag values in bytes. Longer values will be cropped.
+    pub max_tag_value_size: usize,
+}
```

Comment:
Does not need to be `pub` anymore.
```suggestion
/// Config needed to normalize a standalone span.
#[cfg(feature = "processing")]
#[derive(Clone, Debug)]
struct NormalizeSpanConfig {
    /// The time at which the event was received in this Relay.
    pub received_at: DateTime<Utc>,
    /// Allowed time range for transactions.
    pub transaction_range: std::ops::Range<UnixTimestamp>,
    /// The maximum allowed size of tag values in bytes. Longer values will be cropped.
    pub max_tag_value_size: usize,
}
```